app/views/handlers/list_repeating_block.py

```diff
@@ -1,26 +1,27 @@
-from typing import Any
+from functools import cached_property
 
 from flask import url_for
 
 from app.views.handlers.list_edit_question import ListEditQuestion
 
 
 class ListRepeatingBlock(ListEditQuestion):
-    def __init__(self, *args: Any) -> None:
-        super().__init__(*args)
-        self._repeating_block_ids: list[str] = [block["id"] for block in self.parent_block["repeating_blocks"]]
+
+    @cached_property
+    def repeating_block_ids(self) -> list[str]:
+        return [block["id"] for block in self.parent_block["repeating_blocks"]]
```

Comment:
returning a mutable object from a cached property will return a reference rather than a value - you'd need to make this a tuple